doc/contributing/traitsInDepth.md

```diff
@@ -0,0 +1,45 @@
+# Traits in Depth
+
+The Trait system is Terria's way of managing a priority order for state. For example, we might have a default opacity property from a config file that should be overriden by a user provided value. This article will go through how Traits are made, how the priority system works and how updates happen in the UI.
+
+## Making a Trait
+![Making a trait](./img/making_a_trait.png)
+
+For base traits such as `BaseMapsTraits`, we extend ModelTraits. However when composing multiple traits to form a new trait, we use the `mixTraits` function. For example, `ArcGisTerrainCatalogItemTraits` is composed out of `UrlTraits, MappableTraits, CatalogMemberTraits`.
+
+When creating a new property within a Trait, we use [decorators](https://babeljs.io/docs/en/babel-plugin-proposal-decorators) to ensure that the field takes part in the Trait system.
```

Comment:
I'm worried about pluralisation here ("Trait" and "Traits"). Trait only refers to a single value <sub>which could be a single object (and if the type of that trait is `objectTrait` its shape is specified by a class of Traits)</sub>

In this diagram and doc I think most instances of "Trait" should be "Traits". And you wouldn't `mixTraits` on TraitA & TraitB, but you could mix ATraits & BTraits.